### PR TITLE
Fix Storage Drawers Item Handlers issues

### PR DESCRIPTION
**Pull request by PrototypeTrousers**
Fix voiding of items  when nearing full capacity
Fix Slotless Item handler implementation not allowing the extraction from compacting item drawers with the vending upgrade
Cache the Drawer Controller tile to avoid getting the TE from the world every time a drawer slave is interacted with (getSlots,getStackInSlot, etc.)

### CODE REVIEW

**Comment by Rongmario** [comment]
On-hold until we discuss if we should initiate a StorageDrawers fork.

**Comment by Rongmario** [comment]
@PrototypeTrousers we'll pull this in first, but can you resolve the conflicts first?

File: src/main/java/zone/rong/loliasm/common/modfixes/storagedrawers/mixins/ControllerDataMixin.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+    @Inject(method = "getController", at = @At(value = "HEAD"), cancellable = true)
+    void getController(TileEntity host, CallbackInfoReturnable<TileEntityController> cir) {
+        if (controller != null) {
+            if (!controller.isInvalid()) {
+                cir.setReturnValue(controller);
+            }
+            controller = null;
+            host.markDirty();
+            cir.setReturnValue(null);
```
Comment (Rongmario) [inline src/main/java/zone/rong/loliasm/common/modfixes/storagedrawers/mixins/ControllerDataMixin.java:28]: Is this meant to `setReturnValue` twice? I think you have to manually exit the method even when calling `setReturnValue`

File: src/main/java/zone/rong/loliasm/common/modfixes/storagedrawers/mixins/DrawerItemHandlerMixin.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+import javax.annotation.Nonnull;
+
+@Mixin(value = DrawerItemHandler.class, remap = false)
+public class DrawerItemHandlerMixin {
+    @Inject(method = "insertItem", at = @At("HEAD"), cancellable = true)
+    public void insertItem(int slot, ItemStack stack, boolean simulate, CallbackInfoReturnable<ItemStack> cir) {
+        if (slot == 0) {
+            cir.setReturnValue(stack);
+        }
+        cir.setReturnValue(this.insertItemInternal(slot, stack, simulate));
```
Comment (Rongmario) [inline src/main/java/zone/rong/loliasm/common/modfixes/storagedrawers/mixins/DrawerItemHandlerMixin.java:20]: Same as previous point

File: src/main/java/zone/rong/loliasm/config/LoliConfig.java [hunk 3]
```diff
@@ -71,7 +71,7 @@ public boolean shouldSkipClass(Class<?> clazz) {
     public boolean optimizeSomeRendering, stripUnnecessaryLocalsInRenderHelper;
     public boolean quickerEnableUniversalBucketCheck, stripInstancedRandomFromSoundEventAccessor, classCaching, copyScreenshotToClipboard, releaseScreenshotCache;
     public boolean fixBlockIEBaseArrayIndexOutOfBoundsException, cleanupChickenASMClassHierarchyManager, optimizeAmuletRelatedFunctions, labelCanonicalization, skipCraftTweakerRecalculatingSearchTrees, bwmBlastingOilOptimization, optimizeQMDBeamRenderer, repairEvilCraftEIOCompat, optimizeArcaneLockRendering, fixXU2CrafterCrash, disableXU2CrafterRendering;
-    public boolean fixAmuletHolderCapability, delayItemStackCapabilityInit;
+    public boolean fixAmuletHolderCapability, delayItemStackCapabilityInit, fixDrawersItemHandlers;
```
Comment (Rongmario) [inline src/main/java/zone/rong/loliasm/config/LoliConfig.java:74]: Move this to after `disableXU2CrafterRendering` instead as its in the `modfixes` category

File: src/main/java/zone/rong/loliasm/common/modfixes/storagedrawers/mixins/ControllerDataMixin.java (hunk 1 above)
Comment (PrototypeTrousers) [reply src/main/java/zone/rong/loliasm/common/modfixes/storagedrawers/mixins/ControllerDataMixin.java:28]: from what i understood setReturnValue is enough
```
    public void setReturnValue(R returnValue) throws CancellationException {
        super.cancel();
        this.returnValue = returnValue;
    }
```

Comment (embeddedt) [reply src/main/java/zone/rong/loliasm/common/modfixes/storagedrawers/mixins/ControllerDataMixin.java:28]: `setReturnValue` just cancels the rest of the *original* method from running, but your mixin will keep running till you return from it.

**Pull request closed without merging**